### PR TITLE
Added RadioButtonGroup

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Added RadioButtonGroup.
This was needed to make radio button accessibility work. As part of these changes, withFocus was removed from RadioButton, since it wasn't working correctly anyway.

#### Where should the reviewer start?

RadioButtonGroup/doc.js

#### What testing has been done on this PR?

storybook

#### How should this be manually tested?

storybook

#### What are the relevant issues?

https://github.com/grommet/grommet/issues/2676

#### Do the grommet docs need to be updated?

yes

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

Given that focus handling and accessibility was broken in RadioButton, I think we can declare these changes as backwards compatible.

### CODE REVIEW

**Comment by reesemclean** [comment]
This looks like a must! Keyboard navigation seems completely broken for the current RadioButton. I hope this can get merged in as soon as possible.

File: src/js/components/RadioButtonGroup/__tests__/RadioButtonGroup-test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      </Grommet>,
+    );
+    const tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  test('string options', () => {
+    const component = renderer.create(
+      <Grommet>
+        <RadioButtonGroup name="test" options={['one', 'two']} value="one" />
```
Comment (ShimiSun) [inline src/js/components/RadioButtonGroup/__tests__/RadioButtonGroup-test.js]: this test is throwing 
```
PASS  src/js/components/RadioButtonGroup/__tests__/RadioButtonGroup-test.js
  RadioButtonGroup
    ✓ default (33ms)
    ✓ string options (36ms)
    ✓ object options just value (23ms)
    ✓ object options (16ms)
    ✓ object options disabled (18ms)

 › 1 snapshot written.
  console.error node_modules/react/cjs/react.development.js:228
    Warning: Each child in an array or iterator should have a unique "key" prop.

    Check the render method of `RadioButtonGroup`. See https://fb.me/react-warning-keys for more information.
        in WithTheme(RadioButton) (created by RadioButtonGroup)
        in RadioButtonGroup (created by RadioButtonGroup)
        in RadioButtonGroup
        in div (created by Context.Consumer)
        in StyledComponent (created by StyledGrommet)
        in StyledGrommet (created by Grommet)
        in Grommet
```

File: src/js/components/RadioButton/README.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 ## RadioButton
 A radio button control.
 
+RadioButton should typically not be used directly.
```
Comment (ShimiSun) [inline src/js/components/RadioButton/README.md]: should we add an action item to fix the current storybook examples for `RadioButton` to be wrapped with `RadioButtonGroup`?

File: src/js/components/RadioButtonGroup/radiobuttongroup.stories.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    const { value } = this.state;
+    return (
+      <Grommet theme={grommet}>
+        <Box align="center" pad="large">
+          <RadioButtonGroup
+            name="radio"
+            options={[
+              { label: 'Choice 1', value: 'c1' },
+              { label: 'Choice 2', value: 'c2' },
+              { label: 'Choice 3', value: 'c3' },
```
Comment (ShimiSun) [inline src/js/components/RadioButtonGroup/radiobuttongroup.stories.js]: the buttons are not really working on the example, and the focus is always set on the first choice (even when clicking on others).

File: src/js/components/RadioButtonGroup/__tests__/RadioButtonGroup-test.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/RadioButtonGroup/__tests__/RadioButtonGroup-test.js]: good catch! fixed.

**Pull request merged**